src/main/java/org/jenkinsci/plugins/docker/workflow/client/WindowsDockerClient.java

```diff
@@ -0,0 +1,137 @@
+package org.jenkinsci.plugins.docker.workflow.client;
+
+import com.google.common.base.Optional;
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Node;
+import hudson.util.ArgumentListBuilder;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.*;
+import java.nio.charset.Charset;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class WindowsDockerClient extends DockerClient {
+    private static final Logger LOGGER = Logger.getLogger(WindowsDockerClient.class.getName());
+
+    private final Launcher launcher;
+    private final Node node;
+
+    public WindowsDockerClient(@Nonnull Launcher launcher, @CheckForNull Node node, @CheckForNull String toolName) {
+        super(launcher, node, toolName);
+        this.launcher = launcher;
+        this.node = node;
+    }
+
+    @Override
+    public String run(@Nonnull EnvVars launchEnv, @Nonnull String image, @CheckForNull String args, @CheckForNull String workdir, @Nonnull Map<String, String> volumes, @Nonnull Collection<String> volumesFromContainers, @Nonnull EnvVars containerEnv, @Nonnull String user, @Nonnull String... command) throws IOException, InterruptedException {
+        ArgumentListBuilder argb = new ArgumentListBuilder("docker", "run", "-d", "-t");
+        if (args != null) {
+            argb.addTokenized(args);
+        }
+
+        if (workdir != null) {
+            argb.add("-w", workdir);
+        }
+        for (Map.Entry<String, String> volume : volumes.entrySet()) {
+            argb.add("-v", volume.getKey() + ":" + volume.getValue());
```

Comment:
Should we use `--mount` instead?

Reply:
As an aside, I notice the regular docker client uses volumes instead of mounts as well. Should probably keep them consistent (which I suspect means switching DockerClient.java to mounts too mind you - and a separate PR)